Gormsson/Source/GattCharacteristic/TemperatureMeasurement.swift

```diff
@@ -0,0 +1,121 @@
+//  Copyright © 2019 Loïc GRIFFIE. All rights reserved.
+//
+
+import CoreBluetooth
+
+internal final class TemperatureMeasurement: CharacteristicProtocol {
+    public var uuid: CBUUID {
+        return CBUUID(string: "2A1C")
+    }
+
+    public var service: GattService {
+        return .healthThermometer
+    }
+
+    public var format: DataInitializable.Type {
+        return TemperatureMeasurementType.self
+    }
+}
+
+public enum TemperatureMeasurementTypeEnum: UInt8, CustomStringConvertible {
+    case armpit = 1
+    case body
+    case ear
+    case finger
+    case gastroIntestinalTract
+    case mouth
+    case rectum
+    case toe
+    case tympanum
+
+    case unknown
+
+    public var description: String {
+        switch self {
+        case .armpit:
+            return "Armpit"
+        case .body:
+            return "Body (general)"
+        case .ear:
+            return "Ear (usually ear lobe)"
+        case .finger:
+            return "Finger"
+        case .gastroIntestinalTract:
+            return "Gastro-intestinal Tract"
+        case .mouth:
+            return "Mouth"
+        case .rectum:
+            return "Rectum"
+        case .toe:
+            return "Toe"
+        case .tympanum:
+            return "Tympanum (ear drum)"
+        default:
+            return "Reserved for future use"
+        }
+    }
+}
+
+public enum TemperatureMeasurementUnitEnum: UInt8, CustomStringConvertible {
+    case celsius = 0
+    case fahrenheit
+
+    case unknown
+
+    public var description: String {
+        switch self {
+        case .celsius:
+            return "Celsius"
+        case .fahrenheit:
+            return "Fahrenheit"
+        default:
+            return "Reserved for future use"
+        }
+    }
+}
+
+// See: https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.temperature_measurement.xml
+public class TemperatureMeasurementType {
+    private let characteristicData: [UInt8]
+
+    public var value: Float {
+        let bitValue = UInt32(characteristicData[1] << 24) +
+            UInt32(characteristicData[2] << 16) +
+            UInt32(characteristicData[3] << 8) +
+            UInt32(characteristicData[4])
+        return Float(bitPattern: bitValue)
+    }
+
+    public var unit: TemperatureMeasurementUnitEnum {
+        return characteristicData[0] & 0x01 == 0 ? .celsius : .fahrenheit
+    }
+
+    public var type: TemperatureMeasurementTypeEnum {
+        return TemperatureMeasurementTypeEnum.init(rawValue: UInt8(characteristicData[2])) ?? .unknown
+    }
+
+    public var timestamp: Date? {
+        guard characteristicData[0] & 0x2 == 1 else { return nil }
+
+        let dateComponents = DateComponents(calendar: Calendar.current,
+                                            timeZone: TimeZone(secondsFromGMT: 0),
+                                            year: Int(UInt16(characteristicData[5] << 48) + UInt16(characteristicData[6] << 40)),
```

Comment:
5 and 6 is a UInt16 on there own, so it will be just 8 and 0 not 48 and 40.
By the way, I fix this in the coming PR because it should be 0 and 8 because 1 octet reads from right to left but each octets are reads left to right.